ivadomed/main.py

```diff
@@ -362,185 +414,381 @@ def run_command(context, n_gif=0, thr_increment=None, resume_training=False, no_
+    # Run analysis
+    thr = imed_testing.threshold_analysis(model_path=str(model_path),
+                                          ds_lst=[ds_train, ds_valid],
+                                          model_params=model_params,
+                                          testing_params=testing_params,
+                                          metric=metric,
+                                          increment=thr_increment,
+                                          fname_out=str(Path(path_output, "roc.png")),
+                                          cuda_available=cuda_available)
+    # Update threshold in config file
+    context[ConfigKW.POSTPROCESSING][PostprocessingKW.BINARIZE_PREDICTION] = {BinarizeProdictionKW.THR: thr}
+    save_config_file(context, path_output)
+
+
+def execute_traditional_test(bids_df, context, cuda_available, device, loader_params, model_params, path_output,
+                             test_lst, testing_params, transformation_dict):
+    # LOAD DATASET
+    # Warn user that the input-level dropout is set during inference
+    if loader_params[LoaderParamsKW.IS_INPUT_DROPOUT]:
+        logger.warning("Input-level dropout is set during testing. To turn this option off, set 'is_input_dropout'"
+                       "to 'false' in the configuration file.")
+    ds_test = imed_loader.load_dataset(bids_df, **{**loader_params, **{'data_list': test_lst,
+                                                                       'transforms_params': transformation_dict,
+                                                                       'dataset_type': 'testing',
+                                                                       'requires_undo': True}}, device=device,
+                                       cuda_available=cuda_available)
+    eval_params = context[ConfigKW.EVALUATION_PARAMETERS]
+    metric_fns = imed_metrics.get_metric_fns(ds_test.task, eval_params)
+
+    if ModelParamsKW.FILM_LAYERS in model_params and any(model_params[ModelParamsKW.FILM_LAYERS]):
+        ds_test, model_params = update_film_model_params(context, ds_test, model_params, path_output)
+    # RUN INFERENCE
+    pred_metrics = imed_testing.test(model_params=model_params,
+                                     dataset_test=ds_test,
+                                     testing_params=testing_params,
+                                     path_output=path_output,
+                                     device=device,
+                                     cuda_available=cuda_available,
+                                     metric_fns=metric_fns,
+                                     postprocessing=context[ConfigKW.POSTPROCESSING])
+    # RUN EVALUATION
+    df_results = imed_evaluation.evaluate(bids_df, path_output=path_output,
+                                          target_suffix=loader_params[LoaderParamsKW.TARGET_SUFFIX],
                                               eval_params=eval_params)
```

Comment:
Just a heads-up, while the "test" function uses a "dataset" and will be compatible with the V2 loader, the evaluate function still relies heavily on the bids_dataframe and the fact that the dataset is organized as BIDS.

We will have to fix this before having the V2 FILES loader up and running (as phase 5 maybe?)

Reply:
Yeah, correct. I think Metadata and Test function (or most things outside of "training) will need to be in another PR before BIDS Dataset update.